packages/parser/first_block.go

```diff
@@ -108,48 +110,60 @@ func (p FirstBlockParser) Header() *tx.Header {
 	return nil
 }
 
-// FirstBlock generates the first block
-func FirstBlock() {
-	if len(*utils.FirstBlockPublicKey) == 0 {
+// GenerateFirstBlock generates the first block
+func GenerateFirstBlock() error {
+	if len(*conf.FirstBlockPublicKey) == 0 {
 		priv, pub, _ := crypto.GenHexKeys()
-		err := ioutil.WriteFile(*utils.Dir+"/PrivateKey", []byte(priv), 0644)
-		if err != nil {
+		pk := filepath.Join(conf.Config.PrivateDir, "/PrivateKey")
+		if err := ioutil.WriteFile(pk, []byte(priv), 0644); err != nil {
 			log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("writing private key file")
-			return
+			return err
+		}
+		pubk := filepath.Join(conf.Config.PrivateDir, "/PublicKey")
+		if err := ioutil.WriteFile(pubk, []byte(pub), 0644); err != nil {
+			log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("writing public key file")
+			return err
 		}
-		*utils.FirstBlockPublicKey = pub
+		*conf.FirstBlockPublicKey = pub
 	}
-	if len(*utils.FirstBlockNodePublicKey) == 0 {
+
+	if len(*conf.FirstBlockNodePublicKey) == 0 {
 		priv, pub, _ := crypto.GenHexKeys()
-		err := ioutil.WriteFile(*utils.Dir+"/NodePrivateKey", []byte(priv), 0644)
-		if err != nil {
+		nvk := filepath.Join(conf.Config.PrivateDir, "/NodePrivateKey")
+		if err := ioutil.WriteFile(nvk, []byte(priv), 0644); err != nil {
 			log.WithFields(log.Fields{"type": consts.IOError, "error": err}).Error("writing node private key file")
-			return
+			return err
+		}
+		npk := filepath.Join(conf.Config.PrivateDir, "/NodePublicKey")
```

Comment:
the same as above.
Furthermore, pk, pubk, nvk, npk - let's give variables some more different names.

Reply:
Ok.

Reply:
not chaged